perun-engine/src/main/java/cz/metacentrum/perun/engine/runners/SendCollector.java

```diff
@@ -83,9 +87,15 @@ public void run() {
 				 all Destinations (whole Task). Default rescheduleTime is 3 hours * no.of destinations.
 				 */
 				task.setSendEndTime(LocalDateTime.now());
-				if (!Objects.equals(task.getStatus(), Task.TaskStatus.SENDERROR)) {
-					// keep SENDING status only if if task previously doesn't failed
+				// XXX: why is this necessary? Rewriting status with every completed destination? 
+				if (!Objects.equals(task.getStatus(), Task.TaskStatus.SENDERROR) &&
```

Comment:
Not sure, it might be related to the possible race conditions with stuck task processing as with ``sendEndTime`` above. But value assignment for variable shouldn't cost us too much time. We can generally improve this logic in another pull-requests.